src/seedu/addressbook/Main.java

```diff
@@ -72,10 +77,26 @@ private void start(String[] launchArgs) {
 
     /** Prints the Goodbye message and exits. */
     private void exit() {
+    	printTagSessionHistory();
         ui.showGoodbyeMessage();
         System.exit(0);
     }
-
+	
+	/** Prints a concatenated list of tags added/deleted during that session. */
+	private void printTagSessionHistory() {
+		ui.showToUser(DISPLAY_TAG_SESSION_HISTORY);
+		for (Tagging tagging : addressBook.getTagSessionHistory()) {
+			TagAction action = tagging.getAction();
+			String actionSymbol;
+			if (action == TagAction.ADD) {
+				actionSymbol = TAG_SESSION_ADD_ACTION_SYMBOL;
```

Comment:
It is better to have the toString() method of generating a printable string for the Tagging object in the Tagging class itself. Then address book has to just call that method in each Tagging object.